### PR TITLE
Enhancement: Use friendsofphp/php-cs-fixer to enforce PSR-2

### PR DESCRIPTION
**Pull request by localheinz**
This PR
- [x] requires `friendsofphp/php-cs-fixer`
- [x] adds a simple configuration for `php-cs-fixer`, requiring code to comply with PSR-2
- [x] runs `php-cs-fixer` on Travis
- [x] adds a `cs` script to `composer.json`
- [x] runs `composer cs`
- [x] adds a `CONTRIBUTING.md` with instructions

💁 For reference, see https://github.com/FriendsOfPHP/PHP-CS-Fixer.

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+indent_size = 4
+indent_style = space
+insert_final_newline = true
+trim_trailing_whitespace = true
+
+[*.md]
+trim_trailing_whitespace = false
+
+[*.yml]
+indent_size = 2
```
Comment (localheinz) [inline .editorconfig]: This is the indentation as observed in [`.travis.yml`](https://github.com/fastly/fastly-php/blob/master/.travis.yml). [`phpspec/yml`](https://github.com/fastly/fastly-php/blob/master/phpspec.yml), however, is indented with `4` spaces. Was actually going to take a look at it, then noticed the indentation. Hence, this PR.

**Comment by thommahoney** [comment]
I would be interested to see a style guide included in this repo instead of dotfiles for editor configuration. I know that my editor (vim) would not respect this config. At fastly we avoid checking dotfiles into our repos for this reason.

**Comment by localheinz** [comment]
@thommahoney 

How about using [`friendsofphp/php-cs-fixer`](https://github.com/FriendsOfPHP/PHP-CS-Fixer) or [`squizlabs/php_codesniffer`](https://github.com/squizlabs/PHP_CodeSniffer)?

**Comment by localheinz** [comment]
@thommahoney 

Not sure if you've received GitHub notifications, but I have updated this PR to use `friendsofphp/php-cs-fixer` to enforce coding standards. This is probably also the better choice as with `phpspec/phpspec`, `squizlabs/php_codesniffer` will complain a lot about using `snake_case` for method names, and requires custom configuration.

File: src/spec/Fastly/Adapter/Guzzle/GuzzleAdapterSpec.php [hunk 2]
```diff
@@ -7,12 +7,12 @@
 
 class GuzzleAdapterSpec extends ObjectBehavior
 {
-    function it_is_initializable()
+    public function it_is_initializable()
```
Comment (localheinz) [inline src/spec/Fastly/Adapter/Guzzle/GuzzleAdapterSpec.php]: @thommahoney 

About missing visibility, not sure if `phpspec/phpspec` has been updated and provides visibility now, giving this a check.

**Pull request closed without merging**